Mathlib/Analysis/Complex/Tietze.lean

```diff
@@ -0,0 +1,35 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Analysis.Complex.Basic
+import Mathlib.Data.IsROrC.Lemmas
+import Mathlib.Topology.TietzeExtension
+/-!
+# `ℂ` satisfies the Tietze extension theorem
+
+We provide this result here in order to avoid pulling unnecessary imports into either of
+`Topology.TietzeExtension` or `Analysis.Complex.Basic`.
+-/
+
+universe u v w
+
+-- this is not an instance because Lean cannot determine `𝕜`.
+theorem TietzeExtension.of_tvs (𝕜 : Type v) [NontriviallyNormedField 𝕜] {E : Type w}
+    [AddCommGroup E] [Module 𝕜 E] [TopologicalSpace E] [TopologicalAddGroup E] [ContinuousSMul 𝕜 E]
+    [T2Space E] [FiniteDimensional 𝕜 E] [CompleteSpace 𝕜] [TietzeExtension.{u, v} 𝕜] :
+    TietzeExtension.{u, w} E :=
+  Basis.ofVectorSpace 𝕜 E |>.equivFun.toContinuousLinearEquiv.toHomeomorph |> .of_homeo
+
+instance Complex.instTietzeExtension : TietzeExtension ℂ :=
+  TietzeExtension.of_tvs ℝ
+
+instance IsROrC.instTietzeExtension {𝕜 : Type*} [IsROrC 𝕜] : TietzeExtension 𝕜 :=
+  TietzeExtension.of_tvs ℝ
+
+instance IsROrC.instTietzeExtensionTVS {𝕜 : Type v} [IsROrC 𝕜] {E : Type w}
```

Comment:
Shouldn't this be a `(priority := low)` instance?

Reply:
you mean because Lean will always search for `Module ?𝕜 E`? Or because we already have instances for `ℝ` and `ℂ`?

Reply:
As far as I understand, instances that don't have fixed head symbol (e.g., `α × β` or `IsROrC k`, because we only have 2 instances) should have low priority so that Lean apply type-specific instances first (in this case, instances for `Real`, `Complex`, `Prod`, and functions).